app/models/solr_document.rb

```diff
@@ -106,6 +106,28 @@ def iiif_manifest_uri
     iiif_manifest_uris.first
   end
 
+  # Returns the MMS_IDs found in the electronic_access_1display display for URLs that follow the
+  # pattern https://catalog.princeton.edu/catalog\{mms_id}#view except the one for the current ID.
+  # These URLs are found when the Figgy manifest is registered for another (related) MMS_ID rather
+  # than for the current one.
+  def related_bibs_iiif_manifest
+    @related_bibs_iiif_manifest ||= begin
+      string_values = first('electronic_access_1display') || '{}'
+      values = JSON.parse(string_values)
+      mms_ids = values.keys.map { |key| key[/https\:\/\/catalog.princeton.edu\/catalog\/(\d*)#view/, 1] }.compact.uniq
+      mms_ids.map { |id| ensure_alma_id(id) }.select { |mms_id| mms_id != id }
+    end
+  rescue => ex
+    Rails.logger.error "Error calculating related_bibs_iiif_manifest for #{id}: #{ex.message}"
+    []
+  end
+
+  # Makes sure an ID is an Alma ID or converts it to one if it is not.
+  def ensure_alma_id(id)
+    return id if id.length > 7 && id.start_with?("99")
+    "99#{id}3506421"
```

Comment:
Why are you checking id to be > 7 ? Do you consider a potential id of 99106421?
I don't think we need this method since we have in the controller https://github.com/pulibrary/orangelight/blob/f3877565aa40cc3c744728b054531fd4c80a0723/app/controllers/catalog_controller.rb#L17. 
Also if we keep it can you rename it to: ensure_voyager_to_alma_id . An alma id can also end to 06421 if it was created in Alma.

Reply:
Method has been renamed.